modules/va_facilities/app/controllers/va_facilities/v1/nearby_controller.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'will_paginate/array'
+
+require_dependency 'va_facilities/application_controller'
+require_dependency 'va_facilities/pagination_headers'
+require_dependency 'va_facilities/geo_serializer'
+require_dependency 'va_facilities/csv_serializer'
+
+module VaFacilities
+  module V1
+    class NearbyController < ApplicationController
+      include ActionController::MimeResponds
+      include VaFacilities::PaginationHeaders
+      skip_before_action(:authenticate)
+      before_action :set_default_format
+      before_action :validate_params, only: [:index]
+
+      REQUIRED_PARAMS = %i[street_address city state zip].freeze
+      TYPE_SERVICE_ERR = 'Filtering by services is not allowed unless a facility type is specified'
+      MISSING_PARAMS_ERR =
+        'Must supply street_address, city, state, and zip to query nearby facilities.'
+
+      def index
+        resource = NearbyFacility.query(params).paginate(page: params[:page], per_page: params[:per_page])
+        resource = { data: resource, meta: metadata(resource) }
+        respond_to do |format|
+          format.json do
+            render json: resource # ,
+            #  commented out for sending back mocked data that is already in response format
+            #  each_serializer: VaFacilities::FacilitySerializer,
+            #  meta: metadata(resource)
+          end
+          format.geojson do
+            response.headers['Link'] = link_header(resource)
+            render geojson: VaFacilities::GeoSerializer.to_geojson(resource)
+          end
+        end
+      end
+
+      protected
+
+      def set_default_format
+        request.format = :json if params[:format].nil? && request.headers['HTTP_ACCEPT'].nil?
+      end
+
+      private
+
+      def validate_params
+        validate_required_params
+        validate_street_address
+        validate_state_code
+        validate_zip
+        validate_drive_time
+        validate_no_services_without_type
+        validate_type_and_services_known unless params[:type].nil?
+      end
```

Comment:
These look very similar to validations that run against facilities#index. Could we extract them to a mixin when we do the full implementation? Does being on v1 with /nearby mean that's not possible because we want to hold off on that until the other facilities routes increment to v1?

Reply:
Pulling reusable validation methods out into a mixin makes sense to me. I don't see a reason to hold off on doing this until all of facilities is moved over to v1. Most of these validations are common sense and static because of their real world definitions. I have a hard time imagining a change to an address param validator that we wouldn't want to flow to all version of the API.